src/zenml/zen_server/rbac/endpoint_utils.py

```diff
@@ -96,6 +96,48 @@ def verify_permissions_and_create_entity(
     return created
 
 
+def verify_permissions_and_batch_create_entity(
+    batch: List[AnyRequest],
+    resource_type: ResourceType,
+    create_method: Callable[[List[AnyRequest]], List[AnyResponse]],
+) -> List[AnyResponse]:
+    """Verify permissions and create a batch of entities if authorized.
+
+    Args:
+        batch: The batch to create.
+        resource_type: The resource type of the entities to create.
+        create_method: The method to create the entities.
+
+    Raises:
+        IllegalOperationError: If the request model has a different owner then
+            the currently authenticated user.
+        RuntimeError: If the resource type is usage-tracked.
+
+    Returns:
+        The created entities.
+    """
+    auth_context = get_auth_context()
+    assert auth_context
+
+    for request_model in batch:
+        if isinstance(request_model, UserScopedRequest):
+            if request_model.user != auth_context.user.id:
+                raise IllegalOperationError(
+                    f"Not allowed to create resource '{resource_type}' for a "
+                    "different user."
+                )
+
+    verify_permission(resource_type=resource_type, action=Action.CREATE)
```

Comment:
I guess this is not a use case that we might bump into right now, but I am asking this with the future in mind. When we implement these batch-creation endpoints, what happens if only one entity fails during the creation (due to RBAC or any other reason like `EntityExists` errors) Should we fail the whole batch or only exclude the failing ones?

Reply:
For the creation, we only validate that the type is allowed to be created. I think what you have described can not happen.

Reply:
@bcdurak I think the comment on this line is very misleading 😅
As @avishniakov already pointed out, this check is only happening once, and it's happening before any item of the batch is even touched. But I think in general, yes it might happen that item 5 of the batch fails during it's creation.

In that case, we could:
- roll back all previous changes and return a failure for the request
- continue with the next item in the batch, and have the response flexible that it can indicate which items were created and which failed

Currently, we do none of that and simply loop over the batch and once we fail we fail, but keep the already created items in the DB.
I don't think that's ideal, but this PR simply mirrors the previous behaviour for artifact creation at step end, only that it happens with one request. Definitely something to improve on in the future!